src/checks/stylelintConfigIsValid.ts

```diff
@@ -0,0 +1,104 @@
+import { promises as fs } from 'node:fs';
+import { createRequire } from 'node:module';
+import { execaCommand } from 'execa';
+import readdirp from 'readdirp';
+import semver from 'semver';
+import { projectPackageJson } from '../util/packageJson';
+import { supportedFileExtensions } from './stylelint';
+
+const projectDependencies = projectPackageJson.dependencies || {};
+const require = createRequire(`${process.cwd()}/`);
+
+export const title = 'Stylelint config is latest version';
+
+export default async function stylelintConfigIsValid() {
+  // Continue only if project is Next.js or UpLeveled React App
+  if (
+    !(
+      '@upleveled/react-scripts' in projectDependencies ||
+      'next' in projectDependencies
+    )
+  ) {
+    return;
+  }
+
+  const { stdout: remoteVersion } = await execaCommand(
+    'npm show stylelint-config-upleveled version',
+  );
+
+  let localVersion;
+
+  try {
+    const stylelintConfigPackageJsonPath = require.resolve(
+      'stylelint-config-upleveled/package.json',
+    );
+
+    localVersion = JSON.parse(
+      await fs.readFile(stylelintConfigPackageJsonPath, 'utf-8'),
+    ).version;
+  } catch (err) {}
+
+  if (typeof localVersion === 'undefined') {
+    throw new Error(
+      `The UpLeveled Stylelint Config has not been installed. Please install using the instructions on https://www.npmjs.com/package/eslint-config-upleveled
+      `,
+    );
+  }
+
+  if (semver.gt(remoteVersion, localVersion)) {
+    throw new Error(
+      `Your current version of the UpLeveled Stylelint Config (${localVersion}) is older than the latest version ${remoteVersion} - upgrade by running:
+
+      pnpm install stylelint-config-upleveled@${remoteVersion}`,
```

Comment:
is this supposed to be `pnpm add`? maybe check the other Preflight checks that we have

Reply:
7f8a26e